### ISSUE TITLE
FastSpeech2 adaptation

### ISSUE THREAD

**Issue by joanna-aflr** [espnet/espnet#2643]
Hi,

I am trying to fine-tune LJSpeech model (ljspeech_tts_train_conformer_fastspeech2_raw_phn_tacotron_g2p_en_no_space_train.loss.ave) on my dataset. I started the training with this command:

`./run.sh --stage 2 --ngpu 4 --srctexts data/mydata/text --train_set mydata_train --valid_set mydata_deveval --test_sets mydata_deveval --audio_format wav --train_config conf/tuning/train_conformer_fastspeech2.yaml --train_args "--pretrain_path exp/kan-bayashi/ljspeech_conformer_fastspeech2/train.loss.ave_5best.pth --pretrain_key null" --tag finetune_ljspeech_conformer_fastspeech2`

When I try to run inference with some text, the output sounds like speech but I can't understand any word, it doesn't sound at all like the input text. My inference script should be OK because I can generate my text with the pretrained LJSpeech model. Is there anything wrong with my training command?

Thank you,
Joanna

**Comment by kan-bayashi** [comment]
Maybe that is caused by overwriting the token list.
You need to use the token list of the pretrained model.
Please follow https://github.com/espnet/espnet/tree/master/egs2/jvs/tts1#3-replace-token-list-and-statistics-with-pretrained-models-one

**Comment by joanna-aflr** [comment]
Thank you for your reply. The token list is the same as for the pretrained model. Is there anything else you can think of?

**Comment by kan-bayashi** [comment]
How did you get duration?

**Comment by joanna-aflr** [comment]
Ah, yes, the problem was with durations. 
Thank you, I fixed it now, the speech after epoch 2 is already intelligible :) 
One more quick question about adaptation. Should I be using the stats files (`feats_stats.npz`, `energy_stats.npz`, `pitch_stats.npz`) from the pretrained model or the ones generated in stage 5 on my dataset? I had to run stage 5 anyway because I believe `speech_shape` and `text_shape.phn` are needed in training, but I am using the stats files from the pretrained LJSpeech model. Is this the right approach?

Thank you very much for your help!

**Comment by kan-bayashi** [comment]
If you share your problem and solution, it may help other users :)

I also used the pretrained model's statistics but I think it depends on your adaptation dataset size.
If the adaptation dataset size is enough to calculate the statistics, maybe the training with new statistics will work.

**Comment by joanna-aflr** [comment]
Before I just didn't follow FastSpeech2 training procedure to get the durations described here https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/tts1
It worked after I did this. Thanks again!
